contracts/pods/PodAdminGateway.sol

```diff
@@ -18,132 +18,71 @@ import {IPodFactory} from "./interfaces/IPodFactory.sol";
 ///     2. Removing a member from a pod
 ///     3. Transferring a pod member
 ///     4. Toggling a pod membership transfer switch
+///     5. Vetoing a pod proposal
 contract PodAdminGateway is CoreRef, IPodAdminGateway {
     /// @notice Pod factory which creates optimistic pods and acts as a source of information
-    IPodFactory private immutable podFactory;
-
-    constructor(address _core, address _podFactory) CoreRef(_core) {
-        podFactory = IPodFactory(_podFactory);
-    }
-
-    ////////////////////////   GETTERS   ////////////////////////////////
-    // TODO: Whatever global set of roles are, map them one to one to the specific pod roles.
-    // Pod specific roles:
-    // ADMIN: If can add, should be able to remove and lock transfers. Should also be able to veto
-    // GUARDIAN: Can remove and veto 
-    // 
-
-    /// @notice Calculate the specific pod admin role related to adding pod members
-    function getPodAddMemberRole(uint256 _podId)
-        public
-        pure
-        override
-        returns (bytes32)
-    {
-        return keccak256(abi.encode(_podId, "ORCA_POD", "POD_ADD_MEMBER_ROLE"));
-    }
-
-    /// @notice Calculate the pod admin role related to removing pod members
-    function getPodRemoveMemberRole(uint256 _podId)
-        public
-        pure
-        override
-        returns (bytes32)
-    {
-        return
-            keccak256(abi.encode(_podId, "ORCA_POD", "POD_REMOVE_MEMBER_ROLE"));
+    MemberToken public immutable memberToken;
+
+    /// @notice Pod controller for the pods
+    ControllerV1 public immutable podController;
+
+    /// @notice Access control modifier to extend CoreRef ACL to check for 6 roles
+    modifier hasAnyOfSixRoles(
```

Comment:
should be put in coreref

Reply:
I added here as wasn't certain about how we handle applying changes to `CoreRef` and then deploying those. We can add extra functionality and then deploy new contracts with a new `CoreRef` - it's additive only so no need to be concerned about existing contracts having the new functionality available